### PR TITLE
Add a debugging back trace example

### PR DESCRIPTION
**Pull request by jmid**
To improve the debugging guide, this PR adds an example of how to print a back trace for an uncaught exception.
This is briefly mentioned in https://ocaml.org/docs/error-handling#stacktraces too, but I figured a spelled-out example could be useful to a newcomer.

Nit: The error handling page above calls these `stacktraces` (one word) where as the manual page https://v2.ocaml.org/manual/runtime.html calls them both `backtraces` (8 occurrences) and `back traces` (2 occurrences). 

I went with the `back trace` for now
- to side with the manual
- as it seems more correct English, and 
- because it also agrees with `OCAMLRUNPARAM`s `b` option

### CODE REVIEW

**Comment by cuihtlauac** [comment]
This is great @jmid, thanks a lot.

**Review by cuihtlauac — APPROVED** [review]

File: data/tutorials/wf_01_debugging.md [hunk 1]
```diff
@@ -9,12 +9,16 @@ date: 2021-05-27T21:07:30-00:00
 
 # Debugging
 
-This tutorial presents two techniques for debugging OCaml programs:
+This tutorial presents three techniques for debugging OCaml programs:
 
 * [Tracing functions calls](#tracing-functions-calls-in-the-toplevel),
   which works in the interactive toplevel.
```
Comment (christinerose) [inline data/tutorials/wf_01_debugging.md:15]:
```suggestion
  which work in the interactive toplevel.
```

File: data/tutorials/wf_01_debugging.md [hunk 2]
```diff
@@ -339,3 +343,52 @@ Under Emacs you call the debugger using `ESC-x` `ocamldebug a.out`. Then Emacs
 will send you directly to the file and character reported by the debugger, and
 you can step back and forth using `ESC-b` and `ESC-s`, you can set up break
```
Comment (christinerose) [inline data/tutorials/wf_01_debugging.md:344]:
```suggestion
you can step back and forth using `ESC-b` and `ESC-s`. Plus, you can set up break
```

File: data/tutorials/wf_01_debugging.md [hunk 3]
```diff
@@ -339,3 +343,52 @@ Under Emacs you call the debugger using `ESC-x` `ocamldebug a.out`. Then Emacs
...
 points using `CTRL-X space`, and so on...
+
+## Printing a back trace for an uncaught exception
```
Comment (christinerose) [inline data/tutorials/wf_01_debugging.md]:
```suggestion
## Printing A Back Trace for an Uncaught Exception
```

File: data/tutorials/wf_01_debugging.md [hunk 4]
```diff
@@ -339,3 +343,52 @@ Under Emacs you call the debugger using `ESC-x` `ocamldebug a.out`. Then Emacs
...
+
+Getting a back trace for an uncaught exception can be informative to
+understand in which context a problem occurs. However, by default
```
Comment (christinerose) [inline data/tutorials/wf_01_debugging.md]:
```suggestion
understand in which context a problem occurs. However, by default,
```

**Review by christinerose — APPROVED** [review]
Minor formatting suggetions

File: data/tutorials/wf_01_debugging.md (hunk 3 above)
Comment (jmid) [reply data/tutorials/wf_01_debugging.md]: Ah, yes. Good catch!

File: data/tutorials/wf_01_debugging.md (hunk 4 above)
Comment (jmid) [reply data/tutorials/wf_01_debugging.md]: OK :+1:

**Comment by jmid** [comment]
I've made the two corrections relevant to the current PR.

I think the other two suggested text changes are better fit for another PR. In particular, for the first suggestion:
```diff
-  which works in the interactive toplevel.
+  which work in the interactive toplevel.
```
it is the *tracing* (singular) that works in the toplevel, not the 'function calls' (plural).
Hence a different formulation to more clearly convey the intended meaning would be to rephrase it to
```
* [Function call tracing](#tracing-functions-calls-in-the-toplevel),
  which works in the interactive toplevel.
```
Anyway - this is heading into an argument of rephrasing a sentence which is neither authored nor changed in this PR, so I recommend addressing it elsewhere.
